lib/stylelint/plugin.rb

```diff
@@ -1,34 +1,121 @@
 # frozen_string_literal: true
 
+require "mkmf"
+require "json"
+
 module Danger
-  # This is your plugin class. Any attributes or methods you expose here will
-  # be available from within your Dangerfile.
-  #
-  # To be published on the Danger plugins site, you will need to have
-  # the public interface documented. Danger uses [YARD](http://yardoc.org/)
-  # for generating documentation from your plugin source, and you can verify
-  # by running `danger plugins lint` or `bundle exec rake spec`.
+  # Lint javascript files using [stylelint](http://stylelint.io/).
+  # Results are send as inline comment.
   #
-  # You should replace these comments with a public description of your library.
+  # @example Run stylelint with changed files only
   #
-  # @example Ensure people are well warned about merging on Mondays
+  #          stylelint.filtering = true
+  #          stylelint.lint
   #
-  #          my_plugin.warn_on_mondays
-  #
-  # @see  Rossukhon Leagmongkol/danger-stylelint
-  # @tags monday, weekends, time, rattata
+  # @see  nimblehq/danger-stylelint
+  # @tags lint, stylelint, css, scss, sass
   #
   class DangerStylelint < Plugin
-    # An attribute that you can read/write from your Dangerfile
+    DEFAULT_BIN_PATH = "./node_modules/.bin/stylelint"
+
+    # An path to stylelint's config file
+    # @return [String]
+    attr_accessor :config_file
+
+    # An path to stylelint's ignore file
+    # @return [String]
+    attr_accessor :ignore_file
+
+    # Enable filtering
+    # Only show messages within changed files.
+    # @return [Boolean]
+    attr_accessor :filtering
+
+    # A path of stylelint's bin
+    attr_writer :bin_path
+
+    def bin_path
+      @bin_path ||= DEFAULT_BIN_PATH
+    end
+
+    # Specify extensions of target file
+    # Default is [".css, .scss, .sass"]
+    # @return [Array]
+    attr_writer :target_extensions
+
+    def target_extensions
+      @target_extensions ||= %w(.css .scss .sass)
+    end
+
+    # Lints css files.
+    # Generates `errors` and `warnings` according to stylelint's config.
+    #
+    # @return  [void]
+    #
+    def lint
+      lint_results
+        .reject { |result| result.nil? || result["warnings"].length.zero? }
+        .map { |result| send_comment result }
+    end
+
+    private
+
+    # Get stylelint bin path
+    #
+    # return [String]
+    def stylelint_path
+      File.exist?(bin_path) ? bin_path : find_executable("stylelint")
+    end
+
+    # Get all lintable files
+    #
+    # return [String]
+    def all_lintable_files
+      "**/{#{target_extensions.join(',')}}"
```

Comment:
I think based on all lintable files, it needs to be `**/*.{css,scss,sass}`.
Currently i think this returns `**/{.css,.scss,.sass}` instead` 🤔

Reply:
Thanks for noticing, updated in 2baa23eecb06a280aeb23ae84dbf74a2e8332148